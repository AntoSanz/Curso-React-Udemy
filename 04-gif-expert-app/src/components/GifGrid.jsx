import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {

    //TODO: MALA PRACTICA QUE RESOLVER
    getGifs(category);

    return (
        <>
            <h3>{category}</h3>
        </>
    );
};
