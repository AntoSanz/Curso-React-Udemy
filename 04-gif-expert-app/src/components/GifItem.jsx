import PropTypes from 'prop-types';

export const GifItem = ({ title, url} ) => {
    return (
        <>
            <div className="card">
              <img src={url} alt={title}/>
              <p data-testid="test-title">{title}</p>
            </div>
        </>
    );
};

GifItem.PropTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}