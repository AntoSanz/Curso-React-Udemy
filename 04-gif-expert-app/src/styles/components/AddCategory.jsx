import { useState } from 'react';

export const AddCategory = ({ onSetCategories }) => {
    const [inputValue, setInputValue] = useState('Dragon Ball');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length <= 1) {
            return;
        }
        onSetCategories((categories) => [inputValue, ...categories]);
        setInputValue('');
    };

    return (
        <form onSubmit={(e) => onSubmit(e)}>
            <input type="text" placeholder="Buscar GIFs" onChange={(e) => onInputChange(e)} value= {inputValue} />
        </form>
    );
};
