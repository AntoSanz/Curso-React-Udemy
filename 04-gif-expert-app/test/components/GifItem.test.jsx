import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('<GifItem />', () => {
    const myImage = {
        id: 'GRSnxyhJnPsaQy9YLn',
        title: 'Dragon Ball GIF by Toei Animation',
        url: 'https://media2.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=8b52259fu37fe26z6tognzgev27pn0elv5eq3uvar5p8e84w&ep=v1_gifs_search&rid=giphy.gif&ct=g',
    };
    test('should match snapshot', () => {
        // screen.debug();
        const { container } = render(<GifItem title={myImage.title} url={myImage.url} />);
        expect(container).toMatchSnapshot();
    });

    test('should display the image with the "url" filled in', () => {
        render(<GifItem title={myImage.title} url={myImage.url} />);
        const { src } = screen.getByRole('img');
        expect(src).toBe(myImage.url);
    });

    test('should display the image with the "alt" filled in', () => {
        render(<GifItem title={myImage.title} url={myImage.url} />);
        const { alt } = screen.getByRole('img');
        expect(alt).toBe(myImage.title);
    });

    test('should display the title in the <p></p>', () => {
        const { getByTestId } = render(<GifItem title={myImage.title} url={myImage.url} />);
        expect(getByTestId('test-title').innerHTML).toBe(myImage.title);
    });
});
