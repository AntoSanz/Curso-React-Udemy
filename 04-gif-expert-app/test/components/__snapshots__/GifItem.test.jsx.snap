// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifItem /> should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon Ball GIF by Toei Animation"
      src="https://media2.giphy.com/media/GRSnxyhJnPsaQy9YLn/giphy.gif?cid=8b52259fu37fe26z6tognzgev27pn0elv5eq3uvar5p8e84w&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p
      data-testid="test-title"
    >
      Dragon Ball GIF by Toei Animation
    </p>
  </div>
</div>
`;
